### PR TITLE
[ROCm] Fix 2 more unit tests for gpu_kernel_tiling_test on ROCm

### PR DESCRIPTION
**Pull request by i-chaochen**
https://github.com/tensorflow/tensorflow/pull/57948 has fixed one unit test on ColReductionWithSmallDtype.

This PR has fixed rest two (RowReductionTwoRowsPerWarp and RowReductionFourRowsPerWarp) on gpu_kernel_tiling_test at ROCm.

/cc @cheshire

### CODE REVIEW

**Review by cheshire — CHANGES_REQUESTED** [review]
Is it possible to make use of MakePlatformSpecificLlvm and not duplicate the code block?

**Comment by i-chaochen** [comment]
> Is it possible to make use of MakePlatformSpecificLlvm and not duplicate the code block?

Thanks for the feedback. I have removed the duplicate part based on MakePlatformSpecificLlvm.

Hope you're ok with std::string.

**Comment by cheshire** [comment]
@i-chaochen ideally not, maybe we could extract that as well?

**Comment by i-chaochen** [comment]
> @i-chaochen ideally not, maybe we could extract that as well?

Thanks @cheshire for the suggestion on extracting! I have moved all duplicate into MakePlatformSpecificLlvm, now it looks much better.

**Review by cheshire — APPROVED** [review]

**Review by cheshire — APPROVED** [review]

**Comment by i-chaochen** [comment]
@cheshire Apology for asking again, my last changes didn't fix the cuda error. This time I retested on a cuda node without failed, and the code is still following MakePlatformSpecificLlvm with a better extracting to reduce the duplicate.

File: tensorflow/compiler/xla/service/gpu/tests/gpu_codegen_test.cc [hunk 1]
```diff
@@ -68,8 +68,14 @@ std::string GpuCodegenTest::MakePlatformSpecificLlvm(absl::string_view input) {
         is_built_with_rocm_ ? "@llvm.amdgcn.s.barrier" : "@llvm.nvvm.barrier0"},
        {"SHUFFLE", is_built_with_rocm_ ? "i32 @llvm.amdgcn.ds.bpermute"
                                        : "float @llvm.nvvm.shfl.sync.down.f32"},
-       {"TIDX", is_built_with_rocm_ ? "llvm.amdgcn.workitem.id.x"
-                                    : "@llvm.nvvm.read.ptx.sreg.tid.x"}});
+       {"TIDX", is_built_with_rocm_ ? "@llvm.amdgcn.workitem.id.x"
+                                    : "@llvm.nvvm.read.ptx.sreg.tid.x"},
+       {"LCAL", is_built_with_rocm_ ? "%[[LOGICAL_T1:.*]] = call { i1, i64 } @llvm.amdgcn.if.i64(i1 %[[LOGICAL_T0]])"
```
Comment (cheshire) [inline tensorflow/compiler/xla/service/gpu/tests/gpu_codegen_test.cc:73]: Can we merge this one and the one below? It seems they always go together?

**Review by cheshire — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/xla/service/gpu/tests/gpu_codegen_test.cc (hunk 1 above)
Comment (i-chaochen) [reply tensorflow/compiler/xla/service/gpu/tests/gpu_codegen_test.cc:73]: Hmmm....I'm afraid it cannot be. They are two different CHECKs used at different steps for CompilerAndVerifyIr to check. Merge together will change the test order.

**Review by cheshire — APPROVED** [review]

**Pull request merged**